File: boj/python/10828.py
import sys


def input():
    return sys.stdin.readline()


sys.stdin = open('input.txt', 'r')

N = int(input())

queue = []
for i in range(N):
    l = input().rstrip()
    if l[:4] == 'push':
        queue.append(l.split()[1])
    elif l == 'pop':
        if queue:
            print(queue.pop())
        else:
            print(-1)
    elif l == 'size':
        print(len(queue))
    elif l == 'empty':
        if not queue:
            print(1)
        else:
            print(0)
    elif l == 'top':
        if queue:
            print(queue[-1])
        else:
            print(-1)

File: boj/c++/10828.cpp
#include<iostream>
#include<string>
using namespace std;

int stack[10000];
int length=0;
string cmd;

int empty(){
    if (length == 0)
        return 1;
    return 0;
}

void push(int num){
    stack[length++]=num;
}

int pop(){
    if (empty())
        return -1;
    return stack[--length];
}

int size(){
    return length;
}

int top(){
    if (empty())
        return -1;
    return stack[length - 1];
}

int main(){
    int N;
    scanf("%d", &N);
    for (int i = 0; i < N; i++){
        cin >> cmd;
        if (cmd == "push"){
            int num;
            cin >> num;
            push(num);
        }
        else if (cmd == "pop"){
            printf("%d\n", pop());
        }
        else if (cmd == "size"){
            printf("%d\n", size());
        }
        else if (cmd == "empty"){
            printf("%d\n", empty());
        }
        else if (cmd == "top"){
            printf("%d\n", top());
        }
    }
}